src/wifi.rs

```diff
@@ -1050,6 +1086,42 @@ impl<'d> WifiDriver<'d> {
         esp!(unsafe { esp_wifi_set_mac(interface.into(), mac.as_ptr() as *mut _) })
     }
 
+    /// Enable and start WPS
+    pub fn start_wps(&mut self, config: &WpsConfig) -> Result<(), EspError> {
+        let config = Newtype::<esp_wps_config_t>::try_from(config)?;
+
+        match self.get_configuration()? {
```

Comment:
I don't think we should be doing this on behalf of the user. If the chip is not in STA or APSTA mode, the code should just fail.

Reply:
The only problem I see here is that the user will need to call `.set_configuration()` with a ClientConfig in order to be able to start WPS. However, setting a ClientConfig requieres entering WiFi credentials, which are not known at this point. Thus, the user would need to enter some dummy values for SSID and password.

Reply:
Hmmmmm. OK, leave it like this for now.